### PR TITLE
Alerting: Increase size of kvstore value type for MySQL to LONGTEXT

### PR DESCRIPTION
**Pull request by JacobsonMT**
**What is this feature?**

This PR changes the type for kvstore `value` column from `MEDIUMTEXT` to `LONGTEXT`.

**Why do we need this feature?**

Alertmanager uses the kvstore to persist its notification log and the current column limit for MySQL (16.7mb) puts the maximum entries at a level that is potentially achievable for heavy alerting users (~40-80k entries).

In comparison, the current type for PSQL (TEXT) is effectively unlimited and I believe SQLIte defaults to 2gb which is also plenty of leeway.

**Who is this feature for?**

Grafana-managed alerting users with high alert loads.

**Which issue(s) does this PR fix?**:

Reports of errors such as:

```
... msg="unable to create Alertmanager for org" error="error decoding file 'notifications': illegal base64 data at input byte 16777212"
```

**Special notes for your reviewer:**

The nflog is a set of [protobuf messages](https://github.com/prometheus/alertmanager/blob/main/nflog/nflogpb/nflog.proto#L25) that contains some metadata (group key, receiver, group hash) and a list of uint64 hashes for all firing and resolved alert that were in the most recent notification.

Size-wise:
- The size of each entry scales with the cardinality of the firing alert group with some overhead for the metadata. ~M*8 bytes + ~160 bytes (but could be much higher)
- The number of entries scales linearly with the total unique notifications log entries in the past 5 days. Essentially one per (receiver, aggregation group) tuple. N * M

Given `MEDIUMTEXT` has a max size of 16.7 mb and a small entry is ~200-400 bytes, we can estimate the upper bound nflog entries on MySQL at about ~40k-80k.

### CODE REVIEW

**Review by suntala — COMMENTED** [review]
LGTM. We already discussed it and I'm familiar with it so I'll let someone from another team approve it so we have another pair of eyes on this.

File: pkg/services/sqlstore/migrations/kv_store_mig.go [hunk 1]
```diff
@@ -25,3 +25,10 @@ func addKVStoreMigrations(mg *Migrator) {
 
 	mg.AddMigration("add index kv_store.org_id-namespace-key", NewAddIndexMigration(kvStoreV1, kvStoreV1.Indices[0]))
 }
+
+// addKVStoreValueTypeLongTextMigration adds a migration to change the column type of kv_store.value to LONGTEXT for
+// MySQL to be more inline size-wise with PSQL (TEXT) and SQLite.
+func addKVStoreValueTypeLongTextMigration(mg *Migrator) {
+	mg.AddMigration("alter kv_store.value to longtext", NewRawSQLMigration("").
```
Comment (yuri-tceretian) [inline pkg/services/sqlstore/migrations/kv_store_mig.go:32]: I wonder if we want to mention MYSQL in migration name

**Review by yuri-tceretian — APPROVED** [review]
LGTM

File: pkg/services/sqlstore/migrations/kv_store_mig.go (hunk 1 above)
Comment (JacobsonMT) [reply pkg/services/sqlstore/migrations/kv_store_mig.go:32]: Oh boy, I already felt bad with how long it currently is :sweat_smile: but yes I'll add it.

**Pull request merged**